### PR TITLE
Feat/temps de liaison

### PR DESCRIPTION
**Pull request by tristan-gueguen**
Parcours utilisateur:

- On peut activer/désactiver l'option d'entreprise "Temps de liaison"
- Lorsque qu'un gestionnaire review une saisie, il peut ajouter du temps de liaison
- Export PDF depuis historique employé
- Export C1B (comment tester, SOLID ??)
- Export Excel

Impacts:
- Carte Mission
  - Affichage frise de temps / camembert dans la vue "Activités"
  - Calcul/Affichage de l'Amplitude/Temps de travail
- Carte Détails de la journée
  - Calcul/Affichage de l'Amplitude/Temps de travail
 - Seuils Reglementaires
   - 4 regles a verifier / tester

### CODE REVIEW

File: common/utils/mission.js [hunk 1]
```diff
@@ -144,7 +149,7 @@ export function computeMissionStats(m, users) {
       endTime,
       endTimeOrNow,
       service: endTimeOrNow - startTime,
-      totalWorkDuration,
+      totalWorkDuration: totalWorkDuration - transferDuration,
```
Comment (sandrica89) [inline common/utils/mission.js:152]: Are we sure the transferDuration doesn't count in totalWorkDuration?

File: common/utils/regulation/rules.js [hunk 2]
```diff
@@ -69,6 +74,7 @@ function computeTotalWorkDuration(workDayActivities) {
 }
 
 export function checkMaximumDurationOfWork(workDayActivities) {
+  //daily amplitude (includes transfers)
   const totalWorkDuration = computeTotalWorkDuration(workDayActivities);
```
Comment (sandrica89) [inline common/utils/regulation/rules.js:78]: We may rename the variable to know while reading it that it includes transfers

File: web/admin/components/MissionDetails/MissionDetails.js [hunk 3]
```diff
@@ -199,6 +199,9 @@ export function MissionDetails({
       ? missionCompany.settings.requireKilometerData
       : false;
 
+  // not sure about the syntax ??
+  const allowTransfers = missionCompany?.settings?.allowTransfers || false;
```
Comment (sandrica89) [inline web/admin/components/MissionDetails/MissionDetails.js:203]: When I'm not sure, I use unit test.

Remove this comment when it's ok.

File: web/admin/components/MissionDetails/MissionDetails.js [hunk 4]
```diff
@@ -468,6 +471,7 @@ export function MissionDetails({
                               }),
                             handleRevisionAction:
                               missionActions.editSingleActivity,
+                            allowTransfers,
```
Comment (sandrica89) [inline web/admin/components/MissionDetails/MissionDetails.js:474]: No need to add it in `onEditActivity`?

File: web/common/DayRegulationInfo.js [hunk 5]
```diff
@@ -143,15 +143,9 @@ export function DayRegulationInfo({
 
       <RegulationCheck
         key={4}
-        check={
-          checkBreakRespect &&
-          checkBreakRespect.status === RULE_RESPECT_STATUS.failure
-            ? {
-                status: RULE_RESPECT_STATUS.failure,
-                rule: ALERT_TYPES.maximumUninterruptedWorkTime
-              }
```
Comment (sandrica89) [inline web/common/DayRegulationInfo.js:152]: Why was this removed?

File: web/common/Header.js [hunk 6]
```diff
@@ -245,7 +245,7 @@ export function NavigationMenu({ open, setOpen }) {
           )
           .map(route => (
             <ListRouteItem
-              key={route.path || route.href}
+              key={route.path || route.label}
```
Comment (sandrica89) [inline web/common/Header.js:248]: Are we sure `route.label` is unique?

File: web/pwa/components/ActivityRevision/ActivityRevision.js [hunk 7]
```diff
@@ -48,6 +48,7 @@ export default function ActivityRevisionOrCreationModal({
   allowTeamMode = false,
   nullableEndTime = true,
   allowSupportActivity = true,
+  allowTransfers,
```
Comment (sandrica89) [inline web/pwa/components/ActivityRevision/ActivityRevision.js:51]: Shouldn't we use a default value?

File: web/pwa/components/ActivityRevision/ActivityRevision.js [hunk 8]
```diff
@@ -323,6 +324,9 @@ export default function ActivityRevisionOrCreationModal({
               .filter(a =>
                 allowSupportActivity ? true : a !== ACTIVITIES.support.name
               )
+              .filter(a =>
+                allowTransfers ? true : a !== ACTIVITIES.transfer.name
+              )
```
Comment (sandrica89) [inline web/pwa/components/ActivityRevision/ActivityRevision.js:329]: Can't we use only one filter here?

```suggestion
              .filter(a =>
                allowSupportActivity ? true : a !== ACTIVITIES.support.name &&
                allowTransfers ? true : a !== ACTIVITIES.transfer.name
              )
```

File: common/utils/mission.js (hunk 1 above)
Comment (tristan-gueguen) [reply common/utils/mission.js:152]: From my understanding: transfer should impact Amplitude, but not Temps de Travail.
let's say we have 05h-14h(transfer)/14h-15h(drive)
with old version we see amplitude=10h/tempsdetravail=10h in the mission card
with new version we see amplitude=10h/tempsdetravail=1h
which is what I expect

File: web/admin/components/MissionDetails/MissionDetails.js (hunk 4 above)
Comment (tristan-gueguen) [reply web/admin/components/MissionDetails/MissionDetails.js:474]: From my understanding, it is not possible to edit the activity type when editing an activity (only hours for example). Therefore I don't think it's useful to know if we have transfers or not

File: web/common/DayRegulationInfo.js (hunk 5 above)
Comment (tristan-gueguen) [reply web/common/DayRegulationInfo.js:152]: So for this one I think it'd be wise to have a talk with Raphael:
Before, rule 4 was considered automatically broken if rule 3 was broken.
rule 3 is about checking we had sufficient break time during the day (apply only if more than 6h of work)
rule 4 is about checking we don't have more than 6 consecutive hours without a break during the day.
To me it doesn't make sense to say that rule 3 broken => rule 4 broken. That's why I removed this part

File: web/common/Header.js (hunk 6 above)
Comment (tristan-gueguen) [reply web/common/Header.js:248]: No I'm not sure how to solve this issue ?
But I think I understood that `route.href` is undefined, so it was clearly not a good candidate for a key...

File: web/pwa/components/ActivityRevision/ActivityRevision.js (hunk 7 above)
Comment (tristan-gueguen) [reply web/pwa/components/ActivityRevision/ActivityRevision.js:51]: I switched to allowTransfers = false
I don't think I like the idea of needing a default value here though... In theory at least

File: web/pwa/components/ActivityRevision/ActivityRevision.js (hunk 8 above)
Comment (tristan-gueguen) [reply web/pwa/components/ActivityRevision/ActivityRevision.js:329]: Yes we can !

File: web/common/DayRegulationInfo.js (hunk 5 above)
Comment (sandrica89) [reply web/common/DayRegulationInfo.js:152]: To be discussed with @rtaieb

Comment (rtaieb) [reply web/common/DayRegulationInfo.js:152]: I agree with @tristan-gueguen , I don't see why rule 3 was a prerequesite for rule 4.

File: web/admin/components/MissionDetails/MissionDetails.js (hunk 4 above)
Comment (rtaieb) [reply web/admin/components/MissionDetails/MissionDetails.js:474]: In fact it is useful because if you don't put it when you click on "edit activity" for a transfer activity, the label "Liaison" won't be displayed, the field will be blank.

**Review by rtaieb — COMMENTED** [review]
Well done overall :+1: 
Only a small bug when editing transfer activity and other minor comments.

File: web/admin/components/MissionDetails/MissionDetails.js (hunk 4 above)
Comment (tristan-gueguen) [reply web/admin/components/MissionDetails/MissionDetails.js:474]: Ok, added

File: web/admin/components/WorkTimeTable.js [hunk 9]
```diff
@@ -116,6 +116,13 @@ export function WorkTimeTable({
     align: "right",
     minWidth: 100
   };
+  const transferTimeCol = {
+    label: "Liaison",
```
[the remark this answers is not in the document]
Comment (tristan-gueguen) [reply web/admin/components/WorkTimeTable.js:120]: I changed Temp de travail to Travail and Temps de repos to Repos

**Review by rtaieb — APPROVED** [review]
